engine/src/main/java/org/terasology/world/internal/ChunkViewCoreImpl.java

```diff
@@ -211,24 +203,24 @@ public void setExtraData(int index, int blockX, int blockY, int blockZ, int valu
     }
 
     @Override
-    public void setDirtyAround(Vector3i blockPos) {
-        for (Vector3i pos : ChunkMath.getChunkRegionAroundWorldPos(blockPos, 1)) {
-            chunks[pos.x + offset.x + chunkRegion.size().x * (pos.z + offset.z)].setDirty(true);
+    public void setDirtyAround(Vector3ic blockPos) {
+        for (Vector3ic pos : ChunkMath.getChunkRegionAroundWorldPos(blockPos, 1)) {
+            chunks[pos.x() + offset.x + chunkRegion.getSizeX() * (pos.z() + offset.z)].setDirty(true);
         }
     }
 
     @Override
-    public void setDirtyAround(Region3i region) {
-        Vector3i minPos = new Vector3i(region.min());
+    public void setDirtyAround(BlockRegionc region) {
+        Vector3i minPos = region.getMin(new Vector3i());
         minPos.sub(1, 1, 1);
-        Vector3i maxPos = new Vector3i(region.max());
+        Vector3i maxPos = region.getMax(new Vector3i());
         maxPos.add(1, 1, 1);
 
         Vector3i minChunk = ChunkMath.calcChunkPos(minPos, chunkPower);
         Vector3i maxChunk = ChunkMath.calcChunkPos(maxPos, chunkPower);
```

Comment:
Isn't this the same as:

```java
Vector3i tmp = new Vector3i();
BlockRegion expandedRegion = new BlockRegion(region).expand(1, 1, 1);

Vector3i minChunk = ChunkMath.calcChunkPos(expandedRegion.getMin(tmp), chunkPower);
Vector3i maxChunk = ChunkMath.calcChunkPos(expandedRegion.getMax(tmp), chunkPower);
```

This might get a bit esoteric here, but I'd love to write something like:
```java
BlockRegion chunkRegion = region.expand(1, 1, 1).map(corner -> ChunkMath.calcChunkPos(corner, chunkPower);
```

And is there an alternative to `ChunkMath.calcChunkPos(pos, power)` in `Chunks`? I at least don't see one which takes vectors for both arguments when looking at `Chunks.toChunkPos`.

Reply:
On a second look, we introduced `toChunkRegion(BlockRegion region, Vector3ic chunkPower, BlockRegion dest)`, which I guess can (or even should) be used here:

```java
        BlockRegion expandedRegion = new BlockRegion(region).expand(1, 1, 1);
        for (Vector3ic pos : Chunks.toChunkRegion(expandedRegion, chunkPower, expandedRegion) {
            chunks[pos.x() + offset.x + chunkRegion.getSizeX() * (pos.z() + offset.z)].setDirty(true);
        }
```